docs/executor_plugins.md

```diff
@@ -0,0 +1,140 @@
+# Executor Plugins
+
+## Types
+
+### Template Executor
+
+There is only one type of executor plugin, one that runs custom ("plugin") templates, e.g. for non-pod tasks such as
+Tekton builds or Spark jobs.
+
+## A Simple Python Plugin
+
+Let's make a Python plugin that prints "hello" each time the workflow is operated on.
+
+We need the following:
+
+1. [Plugins enabled](plugins.md) .
+2. A HTTP server that will be run as a sidecar to the main container and will respond to RPC HTTP requests from the executor.
+3. Configuration so the controller can discover the plugin.
+
+We'll need to create a script that starts a HTTP server:
+
+```python
+import json
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+
+class Plugin(BaseHTTPRequestHandler):
+
+    def args(self):
+        return json.loads(self.rfile.read(int(self.headers.get('Content-Length'))))
+
+    def reply(self, reply):
+        self.send_response(200)
+        self.end_headers()
+        self.wfile.write(json.dumps(reply).encode("UTF-8"))
+
+    def unsupported(self):
+        self.send_response(404)
+        self.end_headers()
+
+    def do_POST(self):
+        if self.path == '/api/v1/template.execute':
```

Comment:
Need to make sure to document all path types. Also why does the endpoint contain a `.`, is this standard?